src/core/thread/discover_scanner.hpp

```diff
@@ -142,6 +155,11 @@ class DiscoverScanner : public InstanceLocator
         kStateScanDone,
     };
 
+    enum
+    {
+        kMaxLength = 64,
```

Comment:
Suggest renaming this constant to be explicit that it is adv data max length.

Reply:
We are adding two `enum` definitions for the same constant, in here and also in the TLV definition. 
Suggest defining it once in TLV as a public `enum` which then allows it to used by `DiscoverScanner` as well.

Reply:
> Suggest renaming this constant to be explicit that it is adv data max length.

Sure, I can rename it. But just wanted to keep the naming aligned with what's being used everywhere else in the `meshcop_tlvs.hpp`

> We are adding two `enum` definitions for the same constant, in here and also in the TLV definition.
> Suggest defining it once in TLV as a public `enum` which then allows it to used by `DiscoverScanner` as well.

I don't think we can easily share the enum to `discover_scanner.hpp` or other code piece without declaring an object of the TLV class even it's defined as a public member. Do you mean we should make something like `uint8_t mAdvData[MeshCoP::JoinerAdvertisementTlv::kAdvDataMaxLength]`) or am I missing anything?

Reply:
> I don't think we can easily share the enum to `discover_scanner.hpp` or other code piece without declaring an object of the TLV class even it's defined as a public member. Do you mean we should make something like `uint8_t mAdvData[MeshCoP::JoinerAdvertisementTlv::kAdvDataMaxLength])` or am I missing anything?

Yes. We can use any public `enum` constants defined by another `class`/`struct` (key is that it should be public).
So yes, this definition would be valid in `DiscoverScanner`:
```
uint8_t mAdvData[MeshCoP::JoinerAdvertisementTlv::kAdvDataMaxLength])
```

The `enum` definition in this case is a nested type of `MeshCoP::JoinerAdvertisementTlv` class. And using nested types, in general, does not require us to have or declare an object of the enclosing type. The enclosing type is practically acting as a `namespace`.

Reply:
Great, I'll give a try. Thank you Abtin!

Reply:
fixed